modules/heat_transfer/src/relationshipmanagers/GhostBoundary.C

```diff
@@ -99,7 +99,7 @@ bool
 GhostBoundary::operator>=(const RelationshipManager & other) const
 {
   if (auto asoi = dynamic_cast<const GhostBoundary *>(&other))
-    if (_boundary_name == asoi->_boundary_name)
+    if (_boundary_name == asoi->_boundary_name && baseGreaterEqual(*asoi))
       return true;
```

Comment:
this checks the type but it does not check for the boundaries of one being a subset (or a superset) of the other

Reply:
This refers to the solution to an earlier comment. I think your concern should be the topic of a wider/later discussion since other RelationshipManagers, e.g. [(https://github.com/idaholab/moose/blob/next/framework/src/relationshipmanagers/AugmentSparsityOnInterface.C)], don't seem to perform extra checks, so another PR may be in order.

Reply:
this other relationship manager does not take a vector of boundaries/subdomains though? So the check it does truly works to compare the two RMs

Reply:
While you might have spotted a weakness in the current setup I insist this should be part of a follow-up since the issue addressed by this PR was only mildly dependent on a Relationship Manager.

Reply:
I recommend after you approve this PR, since I expect you'd still be displeased, to post an issue with a case where  the error may trickle, and possibly assign it to me.

Reply:
done see https://github.com/idaholab/moose/issues/28330